From: Dan Carpenter <dan.carpenter@linaro.org>
Subject: liveupdate: luo_session: fix use after free in luo_session_deserialize()
Date: Wed, 26 Nov 2025 17:51:40 +0300

The debug output frees "session" but it was freed on the previous line. 
Move the free after the printk().

Link: https://lkml.kernel.org/r/c5dd637d7eed3a3be48c5e9fedb881596a3b1f5a.1764163896.git.dan.carpenter@linaro.org
Signed-off-by: Dan Carpenter <dan.carpenter@linaro.org>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Pratyush Yadav <pratyush@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/liveupdate/luo_session.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/liveupdate/luo_session.c~liveupdate-luo_session-add-sessions-support-fix
+++ a/kernel/liveupdate/luo_session.c
@@ -387,9 +387,9 @@ int luo_session_deserialize(void)
 
 		err = luo_session_insert(sh, session);
 		if (err) {
-			luo_session_free(session);
 			pr_warn("Failed to insert session [%s] %pe\n",
 				session->name, ERR_PTR(err));
+			luo_session_free(session);
 			return err;
 		}
 	}
_
